.github/workflows/build.yml

```diff
@@ -0,0 +1,82 @@
+name: Build
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  CARGO_TERM_COLOR: always
+  RUSTFLAGS: -D warnings
+  RUST_BACKTRACE: 1
+  CARGO_INCREMENTAL: 0
+
+jobs:
+  # build_on_linux:
+  #   name: Build Catalyst Core on Linux
+  #   runs-on: ubuntu-latest
+  #   env:
+  #     CARGO_FLAGS: --verbose --locked
+  #   steps:
+  #     - name: Checkout code
+  #       uses: actions/checkout@v3
+  #       with:
+  #         ref: ${{ github.event.pull_request.head.sha }}
+  #     - uses: Swatinem/rust-cache@v2
+  #     - name: Install Nextest
+  #       run: curl -LsSf https://get.nexte.st/latest/linux | tar zxf - -C ${CARGO_HOME:-~/.cargo}/bin
+  #     - name: Install deps
+  #       run:
+  #         sudo apt install -y protobuf-compiler libssl-dev libpq-dev libsqlite3-dev pkg-config
+
+  #     - name: Build on Linux
+  #       uses: actions-rs/toolchain@v1
+  #       with:
+  #         toolchain: "1.65"
+  #     - run: cargo build
+
+  build_on_windows:
+    name: Build Catalyst Core on Windows
+    runs-on: windows-latest
+    env:
+      CARGO_FLAGS: --verbose --locked
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Allow long paths on Windows
+        shell: powershell
+        run: |
+          reg add HKLM\SYSTEM\CurrentControlSet\Control\FileSystem /v LongPathsEnabled /t REG_DWORD /d 1 /f
+          git config --system core.longpaths true
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Set default linker
+        run: rustup default stable-x86_64-pc-windows-gnu
+
+      - name: Free disk space
+        run: |
```

Comment:
@nicopado @kukkok3 We need to clean up some space on CI for Windows build. Can you help, please?